tests/unit/test_interpolation_tables.py

```diff
@@ -58,47 +59,155 @@ def diff(self, x, axis):
 
 @pytest.mark.parametrize("dim", [2, 3, 4])
 @pytest.mark.parametrize("factory", [_LinearFunc, _QuadraticFunc])
-def test_interpolation(dim, factory):
+def test_value_assigment_and_interpolation(dim, factory):
+    """This test covers all aspects of assignment of values and interpolation using
+    the values. The test is set up to vary the dimension of the parameter space which
+    the tables represents, and the function to be enterpolated.
+
+    It would have been possible to split this into more and smaller tests, however, this
+    would soon have led to code duplication, and it was decided not to pursue this
+    option.
+
+    The general structure of the tests is: For a given function, make full and adaptive
+    interpolation tables. The tables are populated with values (at instantiation for the
+    full table, on demand for the adaptive ones), and the function is interpolated at
+    different points. The accuracy expected depends on the function and on the
+    evaluation points (linear functions should be exactly interpolated, all functions
+    should be exactly interpolated in data points).
+
+    """
+
+    # Set the bounds for the part of the parameter space to be covered by the tables.
+    # The full table will adhere strictly to this,  the adaptive tables are anchored
+    # in the point 'low', and computed according to the resolution of the table (but
+    # since the two types of tables have the same resolution, the resulting grids in
+    # parameter space will have the same nodes).
     low = np.random.rand(dim)
     high = 2 + np.random.rand(dim)
-    resolution = np.array([4 + d for d in range(dim)])
-    max_size = (high - low) / resolution
+    low = np.zeros(dim)
+    high = 2 * np.ones(dim)
+    # Vary the number of points in each dimension.
+    num_pt = np.array([4 + d for d in range(dim)])
+    # Grid resolution in each dimension
+    max_size = (high - low) / (num_pt - 1)
+    # Create function to be tested.
     function = factory(dim, max_size)
 
-    table = pp.InterpolationTable(low, high, resolution, function.func())
-
+    # We have occationally had issues with these tests failing on random data, however,
+    # subsequent testing (on huge arrays) failed to uncover any issues. Fix the random
+    # seed so that if errors arise, they should be reproducible.
+    # TODO: We should rather use hypothesis for this part.
+    np.random.seed(42)
```

Comment:
Despite it being fixed, consider parametrizing the seed.

Reply:
Will do for some of the tests.